napari/_qt/widgets/qt_dims.py

```diff
@@ -130,20 +131,35 @@ def _update_nsliders(self):
 
     def _resize_axis_labels(self):
         """When any of the labels get updated, this method updates all label
-        widths to the width of the longest label. This keeps the sliders
-        left-aligned and allows the full label to be visible at all times,
-        with minimal space, without setting stretch on the layout.
+        widths to a minimum size. This allows the full label to be
+        visible at all times, with minimal space, without setting stretch on
+        the layout.
         """
-        fm = QFontMetrics(QFont("", 0))
-        labels = self.findChildren(QLineEdit, 'axis_label')
-        newwidth = max(fm.boundingRect(lab.text()).width() for lab in labels)
-
-        if any(self._displayed_sliders):
+        displayed_sliders_idx = [
+            idx
+            for idx, displayed in enumerate(self._displayed_sliders)
+            if displayed
+        ]
+        if displayed_sliders_idx:
+            fm = QFontMetrics(QFont("", 0))
```

Comment:
I'm not totally sure 🤔 I just took the way this was being done before to get the `QFontMetrics` instance, but that's a good point. Should I this to use `self.fontMetrics()`?

Reply:
For this PR, I don't need you to since that usage was around anyway, so I've approved.

However, if you're interested into digging into this more in this PR or elsewhere, I won't say no either. Also happy to help debug this by executing stuff on my machine and environments in case there's some dependence there.